code/hwchecker/checker.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python3
+
+import subprocess
+import logging
+import sys
+try:
+    import coloredlogs
+except Exception:
+    print("please install coloredlogs first: pip3 install coloredlogs")
+
+coloredlogs.install()
+
+
+def check(name, commands=None, install=None, function=None):
+    logging.info("Checking {}".format(name))
+    if function is not None:
+        return function()
+    result = subprocess.Popen([commands], shell=True, stdout=subprocess.PIPE)
+    code = result.wait()
+    if code != 0:
+        logging.error("ERR")
+        return install
+    logging.info("OK")
+    return {}
+
+
+def test_lora():
+    try:
+        from SX127x.LoRa import LoRa
+        from SX127x.board_config import BOARD
+    except Exception as e:
+        return {"pip": ["pyLora"]}
+    try:
+        BOARD.setup()
+        lora = LoRa()
+        lora.set_freq(434.2)
+        freq = lora.get_freq()
+        if int(freq) == 0:
+            raise Exception("Not connected")
+        assert 434.15 < freq < 434.25
+        return {}
+    except AssertionError:
+        return {"hardware": ["bad frequency on LoRa radio"]}
+    except Exception:
+        return {"hardware": ["probably no connection to LoRa radio over SPI"]}
+
+
+checks = [
```

Comment:
Can't find pynmea2 check here, it's necessary to efficiently parse GPS data to send it via lora.
Also, I can't find any OLED module being checked here.

Reply:
And fuel gauge 0x36

Reply:
OLED is here: https://github.com/evemorgen/PlasticMonkeysCansat/pull/43/files#diff-7c87f18affdb9493c991e4d3f81a3504R56
Added `pynmea2` and `fuel gauge`